src/agents/supervisor/agent.py

```diff
@@ -1,128 +1,115 @@
-import functools
-import os
-from collections.abc import AsyncGenerator
-from typing import Any, Literal, Protocol
+import json
+from typing import Any, Dict, Literal, Protocol  # noqa UP
 
-from gen_ai_hub.proxy.langchain import ChatOpenAI
-from langchain.agents import AgentExecutor
-from langchain_core.messages import HumanMessage
-from langchain_core.runnables import RunnableConfig
-from langchain_core.tools import tool
-from langfuse.callback import CallbackHandler
-from langgraph.checkpoint import BaseCheckpointSaver
-from langgraph.constants import END
-from langgraph.graph import START, MessagesState, StateGraph
-from langgraph.graph.graph import CompiledGraph
-from pydantic import BaseModel
+from langchain_core.output_parsers import PydanticOutputParser
+from langchain_core.output_parsers.openai_functions import JsonOutputFunctionsParser
+from langchain_core.prompts import ChatPromptTemplate, MessagesPlaceholder
 
-from utils.agents import create_agent
+from agents.common.state import AgentState, Plan
+from agents.common.utils import filter_messages
 from utils.logging import get_logger
+from utils.models import Model
 
-logger = get_logger(__name__)
-
-langfuse_handler = CallbackHandler(
-    secret_key=os.getenv("LANGFUSE_SECRET_KEY"),
-    public_key=os.getenv("LANGFUSE_PUBLIC_KEY"),
-    host=os.getenv("LANGFUSE_HOST"),
-)
-
-
-class Message(BaseModel):
-    """Message data model."""
-
-    input: str
-    session_id: str
-
-
-class Agent(Protocol):
-    """Agent interface."""
-
-    async def astream(self, message: Message) -> dict[str, Any]:
-        """Stream the input to the supervisor asynchronously."""
-        ...
-
-
-@tool
-def search(query: str) -> list[str]:
-    """Call to get information about Kyma."""
-    # This is a placeholder for the actual implementation
-    return [
-        "Kyma is an opinionated set of Kubernetes-based modular building blocks, "
-        "including all necessary capabilities to develop and run "
-        "enterprise-grade cloud-native applications."
-    ]
-
-
-def should_continue(state: MessagesState) -> Literal["action", "__end__"]:
-    """Return the next node to execute."""
-    last_message = state["messages"][-1]
-    return "action" if last_message.tool_calls else "__end__"
+SUPERVISOR = "Supervisor"
+COMMON = "Common"
+FINALIZER = "Finalize"
 
-
-def filter_messages(messages: list) -> list:
-    """This is very simple helper function which only ever uses the last four messages"""
-    return messages[-10:]
-
-
-def agent_node(
-    state: dict[str, Any], agent: AgentExecutor, name: str
-) -> dict[str, Any]:
-    """It filters the messages and invokes the agent."""
-    state["messages"] = filter_messages(state["messages"])
-
-    result = agent.invoke(state)
-    return {"messages": [HumanMessage(content=result["output"], name=name)]}
+logger = get_logger(__name__)
 
 
 class SupervisorAgent:
```

Comment:
Consider renaming to task executor. Then we will merge planner, task executor, replan, and finalizer into the subgraph called SupervisorAgent

Reply:
I would leave this as Supervisor as it eventually will have more functionality. Planning is just temporarily separated out for the start. It will be part of SupervisorAgent too soon.